model/job-management/src/main/proto/beam_construction_api.proto

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Protocol Buffers describing the Job API, api for communicating with a runner
+ * for job submission over GRPC.
+ */
+
+syntax = "proto3";
+
+package org.apache.beam.model.construction.v1;
+
+option go_package = "construction_v1";
+option java_package = "org.apache.beam.model.construction.v1";
+option java_outer_classname = "ConstructionApi";
+
+import "beam_runner_api.proto";
+
+message ConstructionRequest {
+  org.apache.beam.model.pipeline.v1.Components components = 1;
+  org.apache.beam.model.pipeline.v1.PTransform transform = 2;
```

Comment:
Should environment where expansion should happen [1] be part of the expansion request ?

[1] https://github.com/apache/beam/blob/master/model/pipeline/src/main/proto/beam_runner_api.proto#L952

Reply:
These environments refer to runtime environments, which the expander may choose. Possibly it makes sense to pass along some hints/suggestions, though how best to do this in general is unclear. (The choice of expansion service is the choice of environment where the expansion should occur. Such a service could internally be multiplexing, of course.)